src/system/SGX.cpp

```diff
@@ -0,0 +1,64 @@
+
+#ifndef FAASM_SGX
```

Comment:
This file now captures a lot of the SGX-specific logic for set-up and teardown, thus isolating the complexity.